src/garage/_dtypes.py

```diff
@@ -532,19 +574,59 @@ def __new__(cls, env_spec, observation, action, reward, next_observation,
             raise ValueError('env_info must be type {}, but got type {} '
                              'instead.'.format(dict, type(env_info)))
 
-        # rewards
         if not isinstance(reward, float):
             raise ValueError('reward must be type {}, but got type {} '
                              'instead.'.format(float, type(reward)))
 
-        if not isinstance(terminal, bool):
+        if not isinstance(step_type, StepType):
             raise ValueError(
-                'terminal must be dtype bool, but got dtype {} instead.'.
-                format(type(terminal)))
+                'step_type must be dtype garage.StepType, but got dtype {} '
+                'instead.'.format(type(step_type)))
 
         return super().__new__(TimeStep, env_spec, observation, action, reward,
-                               next_observation, terminal, env_info,
-                               agent_info)
+                               next_observation, env_info, agent_info,
+                               step_type)
+
+    def first(self):
+        """Checks if the `TimeStep` is the first one in a sequence.
+
+        Returns:
+            bool: True if it is a FIRST `TimeStep`.
+        """
+        return self.step_type is StepType.FIRST
+
+    def mid(self):
+        """Checks if the `TimeStep` is the mid one in a sequence.
+
+        Returns:
+            bool: True if it is a MID `TimeStep`.
+        """
+        return self.step_type is StepType.MID
+
+    def terminal(self):
```

Comment:
these should be properties

you can document a property like an attribute, e.g.
```python
@property
def terminal(self):
    """ bool: Whether this`TimeStep` is associated with a termination condition."""
    return self.step_type is StepType.TERMINAL
```